nixos/modules/services/misc/gitit.nix

```diff
@@ -0,0 +1,598 @@
+          specified, $HOME/.pandoc will be searched. See pandoc's README for
+          more information.
+        '';
+      };
+
+      xssSanitize = mkOption {
+        type = yesNo;
+        default = "yes";
+        description = ''
+          If yes, all HTML (including that produced by pandoc) is filtered
+          through xss-sanitize.  Set to no only if you trust all of your
+          users.
+        '';
+      };
+  };
+
+  configFile = pkgs.writeText "gitit.conf" ''
+    address: ${cfg.address}
+    port: ${cfg.port}
+    wiki-title: ${cfg.wikiTitle}
+    repository-type: ${cfg.repositoryType}
+    repository-path: ${cfg.repositoryPath}
+    require-authentication: ${cfg.requireAuthentication}
+    authentication-method: ${cfg.authenticationMethod}
+    user-file: ${cfg.userFile}
+    session-timeout: ${cfg.sessionTimeout}
+    static-dir: ${cfg.staticDir}
+    default-page-type: ${cfg.defaultPageType}
+    math: ${cfg.math}
+    mathjax-script: ${cfg.mathJaxScript}
+    show-lhs-bird-tracks: ${cfg.showLhsBirdTracks}
+    templates-dir: ${cfg.templatesDir}
+    log-file: ${cfg.logFile}
+    log-level: ${cfg.logLevel}
+    front-page: ${cfg.frontPage}
+    no-delete: ${cfg.noDelete}
+    no-edit: ${cfg.noEdit}
+    default-summary: ${cfg.defaultSummary}
+    table-of-contents: ${cfg.tableOfContents}
+    plugins: ${cfg.plugins}
+    use-cache: ${cfg.useCache}
+    cache-dir: ${cfg.cacheDir}
+    max-upload-size: ${cfg.maxUploadSize}
+    max-page-size: ${cfg.maxPageSize}
+    debug-mode: ${cfg.debugMode}
+    compress-responses: ${cfg.compressResponses}
+    mime-types-file: ${cfg.mimeTypesFile}
+    use-recaptcha: ${cfg.useReCaptcha}
+    recaptcha-private-key: ${cfg.reCaptchaPrivateKey}
+    recaptcha-public-key: ${cfg.reCaptchaPublicKey}
+    access-question: ${cfg.accessQuestion}
+    access-question-answers: ${cfg.accessQuestionAnswers}
+    rpx-domain: ${cfg.rpxDomain}
+    rpx-key: ${cfg.rpxKey}
+    mail-command: ${cfg.mailCommand}
+    reset-password-message: ${cfg.resetPasswordMessage}
+    use-feed: ${cfg.useFeed}
+    base-url: ${cfg.baseUrl}
+    absolute-urls: ${cfg.absoluteUrls}
+    feed-days: ${cfg.feedDays}
+    feed-refresh-time: ${cfg.feedRefreshTime}
+    pdf-export: ${cfg.pdfExport}
+    pandoc-user-data: ${cfg.pandocUserData}
+    xss-sanitize: ${cfg.xssSanitize}
+  '';
+
+in
+
+{
+
+  options.services.gitit = gititOptions;
+
+  config = mkIf cfg.enable {
+
+    environment.systemPackages = with pkgs; [
+      git
+      (if cfg.pdfExport == "yes" then texLive else null)
+    ];
+
+    users.extraUsers.gitit = {
+      group = config.users.extraGroups.gitit.name;
+      description = "Gitit user";
+      home = "/var/lib/gitit";
+      createHome = true;
+      uid = config.ids.uids.gitit;
+    };
+
+    users.extraGroups.gitit.gid = config.ids.gids.gitit;
+
+    systemd.services.gitit = {
+      description = "The git and pandoc powered wiki";
```

Comment:
Systemd service descriptions should be capitalized (each word). (And I'd lose the 'The'.)